File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/crypto/MlDsaPublicKeyAdmission.java
package org.hyperledger.iroha.android.crypto;

/** Protocol-shape admission checks for raw ML-DSA-65 public keys. */
public final class MlDsaPublicKeyAdmission {

  /** Canonical raw ML-DSA-65 public-key length. */
  public static final int PUBLIC_KEY_LENGTH = 1952;

  private MlDsaPublicKeyAdmission() {}

  /** Returns {@code true} only for an exact-width, nonzero ML-DSA-65 public key. */
  public static boolean isValid(final byte[] publicKey) {
    if (publicKey == null || publicKey.length != PUBLIC_KEY_LENGTH) {
      return false;
    }
    for (final byte value : publicKey) {
      if (value != 0) {
        return true;
      }
    }
    return false;
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/crypto/MlDsaPublicKeyAdmission.kt
package org.hyperledger.iroha.sdk.crypto

/** Structural admission checks for protocol ML-DSA-65 public keys. */
object MlDsaPublicKeyAdmission {
    /** Canonical raw ML-DSA-65 public-key length. */
    const val PUBLIC_KEY_LENGTH: Int = 1_952

    /** Returns `true` only for exact-length, nonzero ML-DSA-65 public-key material. */
    @JvmStatic
    fun isValid(publicKey: ByteArray?): Boolean =
        publicKey != null &&
            publicKey.size == PUBLIC_KEY_LENGTH &&
            publicKey.any { it.toInt() != 0 }
}
